tests/components/Microsoft.DotNet.UpgradeAssistant.MSBuild.Tests/MSBuildTestBuild.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+
+namespace Microsoft.DotNet.UpgradeAssistant.MSBuild.Tests
+{
+    public sealed class MSBuildTestBuild : IDisposable
+    {
+        public string TestFolderPath { get; }
+
+        public MSBuildTestBuild(string path)
```

Comment:
Do tests need to care about the path? Can this be an implementation detail?

Reply:
It should be something that can be done both thread-safely and process safely (this is neither).